File: Python/01_02_Password_Generator.py
# Beginner Project
# A random password generator
# Create a random password for a user

import random


def generate_password(pass_len) :
    try :
        char_str = '1234567890ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz!@#$%^&*_-+'
        new_pass = ''
        for i in range(int(pass_len)) :
            char_len = len(char_str)
            rand_num = random.randint(0, char_len-1)
            get_char = char_str[rand_num]
            new_pass += get_char
        return new_pass

    except Exception as err_ :
        err_message_ = err_.args
        return err_message_


try:
    password_length = input('Enter password length : ')
    print(generate_password(password_length))
except Exception as err :
    err_message = err.args
    print(err_message)

File: CSharp/01_02_Password_Generator/01_02_Password_Generator.cs
// Beginner Project
// A random password generator
// Create a random password for a user


using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CS_Code_Project
{
    internal class Password_Generator
    {

        private const int char_minim = 8;
        private const string charSet = "1234567890ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz!@#$%^&*_-+";
       
        Random random = new Random();
        StringBuilder passwordBuilder = new StringBuilder();
        private int GetPasswordLength()
        {
            while (true)
            {
                Console.WriteLine("Enter the desired length of the password: ");
                string user_input = Console.ReadLine();

                if(int.TryParse(user_input, out int length) && length > char_minim)
                {
                    return length;
                }
                else
                {
                    Console.WriteLine($"Please enter a valid integer greater than or equal to {char_minim}.");
                }
            }
        }

        private string generate_password(int length)
        {
            
            try
            {
                for (int i = 0; i < length; i++)
                {
                    int index = random.Next(charSet.Length);
                    passwordBuilder.Append(charSet[index]);
                }

                return passwordBuilder.ToString();
            }
            catch(Exception err)
            {
                Console.WriteLine(err.Message);
                Console.ReadLine();
            }

            return null;
        }

        static void Main(string[] args)
        {
            Password_Generator generator = new Password_Generator();

            int passwordln = generator.GetPasswordLength();
            string new_pass = generator.generate_password(passwordln);
            Console.WriteLine($"Generated Password: {new_pass}");
            Console.WriteLine("Press Enter to exit.");
            Console.ReadLine();
        }
    }
}
